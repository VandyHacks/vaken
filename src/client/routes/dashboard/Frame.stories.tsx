import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { MemoryRouter } from 'react-router-dom';
import { MyStatusDocument, MyStatusQueryResult } from '../../generated/graphql';
import Component from './Frame';

export default {
	title: 'Routes/Dashboard/Frame',
	component: Component,
} as Meta;

const mocks = [
	{
		request: {
			query: MyStatusDocument,
		},
		result: {
			data: {
				me: {
					id: 'foo',
					status: 'SUBMITTED',
				},
			},
		} as MyStatusQueryResult,
	},
] as MockedResponse[];

export const Frame: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={mocks}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);
