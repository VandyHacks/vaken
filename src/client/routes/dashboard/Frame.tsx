import React, { useState, Suspense, FunctionComponent, useContext } from 'react';
import styled from 'styled-components';
// import { useImmer } from 'use-immer';
import { Switch, Redirect, Route } from 'react-router-dom';
import Sidebar from '../../components/Sidebar/Sidebar';
import STRINGS from '../../assets/strings.json';
import Title from '../../components/Text/Title';
import { routes } from '../../assets/routes';
import { AuthContext } from '../../contexts/AuthContext';
import { SpaceBetweenRow, OverflowContainer } from '../../components/Containers/FlexContainers';
import { ActionButtonContext } from '../../contexts/ActionButtonContext';

export const OrganizerDash = React.lazy(() => import('./OrganizerDash'));

/* eslint-env browser */

const Layout = styled.div`
	height: 100vh;
	width: 100vw;
	display: grid;
	grid:
		'sidebar . . .' 1.5rem
		'sidebar . header .' auto
		'sidebar . . .' 1.5rem
		'sidebar . content .' 1fr
		'sidebar . . .' 1.5rem
		/ 22rem 2rem 1fr 2rem;
	/* align-items: stretch; */
	overflow: hidden;

	.content {
		grid-area: content;
		overflow: auto;
		max-height: 100%;
		/* border-radius: 2rem; */
		overflow: auto;
	}

	.header {
		grid-area: header;
	}
`;

const Rectangle = styled.div`
	height: 0.4rem;
	width: 7.5rem;
	background: ${STRINGS.ACCENT_COLOR};
`;

const Frame: FunctionComponent = (props): JSX.Element => {
	const currentUser = useContext(AuthContext);
	const [ActionButton, setActionButton] = useState<JSX.Element | undefined>(undefined);
	if (window.location.pathname.startsWith('/login')) {
		return <Redirect to="/dashboard" />;
	}

	return (
		<ActionButtonContext.Provider value={{ ActionButton, update: setActionButton }}>
			<Layout>
				<div className="header">
					<SpaceBetweenRow>
						<Title color={STRINGS.ACCENT_COLOR} margin="1.5rem 0rem 0rem">
							<Switch>
								{routes.map(route => {
									return route.authLevel.includes(currentUser.authLevel) ? (
										<Route key={route.path} path={route.path} render={() => route.displayText} />
									) : null;
								})}
							</Switch>
						</Title>
						{ActionButton || null}
					</SpaceBetweenRow>
					<Rectangle />
				</div>
				<Sidebar />
				<OverflowContainer className="content">
					<Suspense fallback={<div>Loading...</div>}>
						<Switch>
							{routes.map(route => {
								return route.authLevel.includes(currentUser.authLevel) ? (
									<Route key={route.path} path={route.path} render={() => <route.component />} />
								) : null;
							})}
						</Switch>
					</Suspense>
				</OverflowContainer>
			</Layout>
		</ActionButtonContext.Provider>
	);
};

export default Frame;
