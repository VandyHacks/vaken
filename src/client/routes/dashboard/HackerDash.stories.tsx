import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { MemoryRouter } from 'react-router-dom';
import { MyStatusDocument, MyStatusQueryResult, ApplicationStatus } from '../../generated/graphql';
import Component from './HackerDash';

export default {
	title: 'Routes/Dashboard/Hacker Dash',
	component: Component,
} as Meta;

function makeMocks(status: ApplicationStatus): MockedResponse[] {
	return [
		{
			request: {
				query: MyStatusDocument,
			},
			result: {
				data: {
					me: {
						id: 'foo',
						status,
						__typename: 'Hacker',
					},
				},
			} as MyStatusQueryResult,
		},
	];
}

export const Created: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Created)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Started: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Started)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Submitted: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Submitted)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Confirmed: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Confirmed)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Accepted: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Accepted)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Declined: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Declined)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);

export const Rejected: Story<Record<string, unknown>> = args => (
	<MockedProvider mocks={makeMocks(ApplicationStatus.Rejected)}>
		<MemoryRouter>
			<Component {...args} />
		</MemoryRouter>
	</MockedProvider>
);
