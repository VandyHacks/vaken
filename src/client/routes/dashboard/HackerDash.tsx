import React, { FunctionComponent, useState, useEffect, RefObject } from 'react';
import styled from 'styled-components';
import { Button } from '../../components/Buttons/Button';
import FloatingPopup from '../../components/Containers/FloatingPopup';
import { FlexColumn, FlexStartColumn } from '../../components/Containers/FlexContainers';
import { Title } from '../../components/Text/Title';
import STRINGS from '../../assets/strings.json';
import { ButtonOutline, CenterButtonText } from '../../components/Buttons/Buttons';
import applicationStatusSVG from '../../assets/img/application_status.svg';
import { SmallCenteredText } from '../../components/Text/SmallCenteredText';
import {
	ApplicationStatus,
	useMyStatusQuery,
	useConfirmMySpotMutation,
	useDeclineMySpotMutation,
} from '../../generated/graphql';

const Confetti = React.lazy(() => import('react-confetti'));

const statusConfig = {
	[ApplicationStatus.Created]: {
		actions: [
			{
				action: () => void (window.location.href = '/application'),
				actionText: 'Complete your application',
			},
		],
		boldText: "You haven't started your application yet.",
		img: applicationStatusSVG,
		status: 'Not Started',
		statusBG: STRINGS.APPLICATION_INCOMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_INCOMPLETE_STATUSCOLOR,
		text: `The deadline is ${STRINGS.DEADLINE}`,
	},
	[ApplicationStatus.Started]: {
		actions: [
			{
				action: () => void (window.location.href = '/application'),
				actionText: 'Complete your application',
			},
		],
		boldText: 'You still need to finish your application.',
		img: applicationStatusSVG,
		status: 'Incomplete',
		statusBG: STRINGS.APPLICATION_INCOMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_INCOMPLETE_STATUSCOLOR,
		text: `The deadline is ${STRINGS.DEADLINE}`,
	},
	[ApplicationStatus.Submitted]: {
		actions: [
			{
				action: () => void (window.location.href = '/application'),
				actionText: 'Update your application',
			},
		],
		boldText: "Thanks for applying! We'll get back to you with a decision shortly.",
		img: applicationStatusSVG,
		status: 'Submitted',
		statusBG: STRINGS.APPLICATION_COMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_COMPLETE_STATUSCOLOR,
		text: "You may update your responses at any time by re-visiting the application.'",
	},
	[ApplicationStatus.Confirmed]: {
		actions: [
			{
				action: () => void window.open('https://discord.gg/MbbfBWW', '_blank'),
				actionText: 'Join our Discord',
			},
		],
		boldText: `Whoo hoo! We'll see you ${STRINGS.START_DAY}! Don't forget to join the Discord!`,
		img: applicationStatusSVG,
		status: 'Confirmed',
		statusBG: STRINGS.APPLICATION_COMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_COMPLETE_STATUSCOLOR,
		text: "If you don't have a team, you can form one when you join the Discord!",
	},
	[ApplicationStatus.Accepted]: {
		actions: [
			{
				action: () => void undefined, // Overridden in HackerDash component
				actionText: 'Confirm your spot',
			},
			{
				action: () => void undefined, // Overridden in HackerDash component
				actionText: 'Decline your spot',
			},
		],
		boldText: "You've been accepted!",
		img: applicationStatusSVG,
		status: 'Accepted',
		statusBG: STRINGS.APPLICATION_COMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_COMPLETE_STATUSCOLOR,
		text: "Confirm your spot to let us know you'll be coming",
	},
	[ApplicationStatus.Declined]: {
		actions: [],
		boldText: "You've declined.",
		img: applicationStatusSVG,
		status: 'Declined',
		statusBG: STRINGS.APPLICATION_DECLINED_STATUSBG,
		statusColor: STRINGS.APPLICATION_DECLINED_STATUSCOLOR,
		text: `Changed your mind? Email ${STRINGS.HELP_EMAIL}`,
	},
	[ApplicationStatus.Rejected]: {
		actions: [],
		boldText: "Unfortunately, we couldn't offer you a spot this year :(",
		img: applicationStatusSVG,
		status: 'Denied',
		statusBG: STRINGS.APPLICATION_INCOMPLETE_STATUSBG,
		statusColor: STRINGS.APPLICATION_INCOMPLETE_STATUSCOLOR,
		text: '',
	},
};

const HackerDashBG = styled(FloatingPopup)`
	border-radius: 8px;
	height: min-content;
	width: 36rem;
	background: rgba(247, 245, 249, 1);
	padding: 1.5rem;

	@media screen and (max-width: 456px) {
		width: 100%;
	}

	@media screen and (max-width: 400px) {
		a {
			width: 100%;
		}
	}

	img {
		display: block;
		height: 200px;
	}
`;

export const HackerDash: FunctionComponent = (): JSX.Element => {
	const { data, loading } = useMyStatusQuery();
	const [statusInfo, setStatusInfo] = useState(statusConfig[ApplicationStatus.Created]);
	const [confirmMySpot] = useConfirmMySpotMutation();
	const [declineMySpot] = useDeclineMySpotMutation();
	const [showConfetti, setShowConfetti] = useState(false);

	useEffect((): void => {
		if (data && data.me && data.me.__typename === 'Hacker') {
			console.log(data.me.status);
			setStatusInfo(statusConfig[data.me.status as keyof typeof statusConfig]);
		}
	}, [data]);

	useEffect((): void => {
		statusConfig[ApplicationStatus.Accepted].actions[0].action = () => {
			confirmMySpot()
				.then(() => setShowConfetti(true))
				.catch(err => console.error(err));
			return undefined;
		};
		statusConfig[ApplicationStatus.Accepted].actions[1].action = () => {
			declineMySpot()
				// no confetti :(
				.catch(err => console.error(err));
			return undefined;
		};
	}, [confirmMySpot, declineMySpot]);

	return (
		<>
			<FlexStartColumn>
				{showConfetti ? (
					<Confetti
						recycle={false}
						numberOfPieces={800}
						canvasRef={(null as unknown) as RefObject<HTMLCanvasElement>}
					/>
				) : null}
				<HackerDashBG>
					<FlexColumn>
						<Title fontSize="1.75rem">{STRINGS.HACKER_DASHBOARD_HEADER_TEXT}</Title>
						{loading ? null : (
							<>
								<ButtonOutline
									paddingLeft="4rem"
									paddingRight="4rem"
									width="auto"
									background={statusInfo.statusBG}
									glowColor="null">
									<CenterButtonText color={statusInfo.statusColor} fontSize="1.8rem">
										{statusInfo.status}
									</CenterButtonText>
								</ButtonOutline>
								<img src={statusInfo.img} alt={`${STRINGS.APPLICATION_STATUS_SVG_ALT_TEXT}`} />
								<SmallCenteredText
									color={`${STRINGS.DARK_TEXT_COLOR}`}
									fontSize="1rem"
									margin="1.4rem">
									<>
										<p style={{ fontWeight: 'bold' }}>{statusInfo.boldText}</p>
										<br />
										{statusInfo.text}
									</>
								</SmallCenteredText>
							</>
						)}
						{statusInfo.actions.map(e => (
							<Button key={e.actionText} large long onClick={e.action}>
								{e.actionText}
							</Button>
						))}
					</FlexColumn>
				</HackerDashBG>
			</FlexStartColumn>
		</>
	);
};

export default HackerDash;
