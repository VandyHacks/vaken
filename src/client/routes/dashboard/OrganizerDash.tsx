import React, { FunctionComponent } from 'react';
import { Bar, Pie } from 'react-chartjs-2';
import { Link } from 'react-router-dom';
import { gql } from 'apollo-boost';
import { useQuery } from 'react-apollo-hooks';
import styled from 'styled-components';
import { ChartOptions } from 'chart.js';

import { Spinner } from '../../components/Loading/Spinner';
import { GraphQLErrorMessage } from '../../components/Text/ErrorMessage';
import FloatingPopup from '../../components/Containers/FloatingPopup';
import TextButton from '../../components/Buttons/TextButton';
import {
	OverflowContainer,
	FlexRow,
	FlexColumn,
	FlexStartColumn,
} from '../../components/Containers/FlexContainers';
import 'chartjs-plugin-datalabels';
import STRINGS from '../../assets/strings.json';

const Label = styled('span')`
	font-size: 1.25rem;
	font-family: 'Roboto', sans-serif;
	font-weight: 500;
	color: ${STRINGS.DARK_TEXT_COLOR};
`;

const Value = styled('span')`
	font-size: 1.25rem;
	font-family: 'Roboto', sans-serif;
	font-weight: 100;
	color: ${STRINGS.DARK_TEXT_COLOR};
`;

const StyledUL = styled.ul`
	font-size: 1rem;
`;

const StyledLI = styled.li`
	margin-bottom: 0.5rem;
`;

const StyledTable = styled.div`
	grid-area: table;
`;

const BarLayout = styled.div`
	grid-area: chart;
`;
const LinkLayout = styled(FlexStartColumn)`
	grid-area: link;
`;

const PieLayoutLeft = styled.div`
	grid-area: pie1;
`;

const PieLayoutRight = styled.div`
	grid-area: pie2;
`;

const StyledFloatingPopupTop = styled(FloatingPopup)`
	display: grid;
	grid-template-columns: 70% 2rem auto;
	grid-template-rows: auto auto;
	grid-template-areas:
		'chart . table'
		'link link link';
`;

const StyledFloatingPopupBottom = styled(FloatingPopup)`
	display: grid;
	grid-template-columns: 50% 50%;
	grid-template-rows: auto;
	grid-template-areas: 'pie1 pie2';
`;

export const GET_STATISTICS = gql`
	query Statistics($number: Float!) {
		getAllHackerGenders {
			Male
			Female
			Other
			PreferNotToSay
		}
		getAllHackerSizes {
			UXS
			US
			UM
			UL
			UXL
			UXXL
			WS
			WM
			WL
			WXL
			WXXL
		}
		getAllHackerStatuses {
			Created
			Verified
			Started
			Submitted
			Accepted
			Confirmed
			Rejected
		}
		getTopHackerSchools(number: $number) {
			school
			counts
		}
	}
`;

const colorPalette = STRINGS.COLOR_PALETTE.slice(1);

const generateColor = (n: number): string[] =>
	[...Array(n).keys()].map((i: number) => colorPalette[i % colorPalette.length]);

const barStatusData = (data: { [key: string]: number }) => {
	const statusData = Object.values(data).slice(0, -1);
	const statusLabels = Object.keys(data).slice(0, -1);
	return {
		datasets: [
			{
				backgroundColor: generateColor(statusData.length),
				data: statusData,
			},
		],
		labels: statusLabels,
	};
};

const barStatusOptions: ChartOptions = {
	legend: {
		display: false,
	},
	maintainAspectRatio: true,
	plugins: {
		datalabels: {
			align: 'start',
			anchor: 'end',
			clip: true,
			color: 'black',
			display: 'auto',
			font: {
				size: 20,
				weight: 'bold',
			},
		},
	},
	responsive: true,
	scales: {
		xAxes: [
			{
				ticks: {
					fontSize: 20,
				},
			},
		],
	},
	showLines: true,
	title: {
		display: true,
		fontSize: 24,
		text: 'Number of Applicants',
	},
	tooltips: {
		enabled: false,
	},
};

const pieShirtData = (data: any) => {
	const shirtData = Object.values(data).slice(0, -1);
	const shirtLabels = Object.keys(data).slice(0, -1);
	return {
		datasets: [
			{
				backgroundColor: generateColor(shirtData.length),
				data: shirtData,
			},
		],
		labels: shirtLabels,
	};
};

const pieShirtOptions = {
	legend: {
		position: 'right' as 'right',
	},
	maintainAspectRatio: true,
	responsive: true,
	title: {
		display: true,
		fontSize: 24,
		position: 'bottom' as 'bottom',
		text: 'T-Shirt Sizes',
	},
};

const pieGenderData = (data: any) => {
	const genderData = Object.values(data).slice(0, -1);
	const genderLabels = Object.keys(data).slice(0, -1);
	return {
		datasets: [
			{
				backgroundColor: generateColor(genderData.length),
				data: genderData,
			},
		],
		labels: genderLabels,
	};
};

const pieGenderOptions = {
	legend: {
		position: 'right' as 'right',
	},
	maintainAspectRatio: true,
	responsive: true,
	title: {
		display: true,
		fontSize: 24,
		position: 'bottom' as 'bottom',
		text: 'Gender',
	},
};

interface Props {
	data: [{ school: string; counts: number }];
}

export const SchoolTable: FunctionComponent<Props> = (props: Props): JSX.Element => {
	const { data } = props;

	return (
		<StyledUL>
			{data.map(d => (
				<StyledLI key={d.school}>
					<Label>{`${d.school}: `}</Label>
					<Value>{d.counts}</Value>
				</StyledLI>
			))}
		</StyledUL>
	);
};

export const OrganizerDash: FunctionComponent = (): JSX.Element => {
	const { loading, error, data } = useQuery(GET_STATISTICS, {
		variables: { number: 5.0 },
	});

	if (loading) return <Spinner />;
	if (error) {
		console.log(error);
		return <GraphQLErrorMessage text={STRINGS.GRAPHQL_ORGANIZER_ERROR_MESSAGE} />;
	}

	return (
		<OverflowContainer>
			<StyledFloatingPopupTop marginBottom="1rem" backgroundOpacity="1" padding="1.5rem">
				<BarLayout>
					<Bar data={barStatusData(data.getAllHackerStatuses)} options={barStatusOptions} />
				</BarLayout>
				<StyledTable>
					<SchoolTable data={data.getTopHackerSchools} />
				</StyledTable>
				<LinkLayout>
					<Link style={{ textDecoration: 'none' }} to="/managehackers">
						<TextButton
							color="white"
							fontSize="1.4em"
							background={STRINGS.ACCENT_COLOR}
							text="Manage hackers"
							glowColor="rgba(0, 0, 255, 0.67)"
						/>
					</Link>
				</LinkLayout>
			</StyledFloatingPopupTop>
			<StyledFloatingPopupBottom backgroundOpacity="1" padding="1.5rem">
				<PieLayoutLeft>
					<Pie data={pieShirtData(data.getAllHackerSizes)} options={pieShirtOptions} />
				</PieLayoutLeft>
				<PieLayoutRight>
					<Pie data={pieGenderData(data.getAllHackerGenders)} options={pieGenderOptions} />
				</PieLayoutRight>
			</StyledFloatingPopupBottom>
		</OverflowContainer>
	);
};

export default OrganizerDash;
