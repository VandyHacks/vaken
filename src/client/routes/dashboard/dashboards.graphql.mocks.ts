/**
 * This file contains mock responses for the queries defined in the corresponding `.graphql.ts`
 * file. It is included in `.storybook/preview.js` to provide default responses for every graphql
 * query in Storybook story files.
 */
import { MockedResponse } from '@apollo/client/testing';
import { MyStatusDocument, MyStatusQueryResult, ApplicationStatus } from '../../generated/graphql';
import { MOCK_HACKER } from '../../../common/mockObjects';

export const MyStatusQueryMock: MockedResponse<MyStatusQueryResult['data']> = {
	request: {
		query: MyStatusDocument,
	},
	result: {
		data: {
			me: {
				id: MOCK_HACKER.id,
				status: ApplicationStatus.Submitted,
				__typename: 'Hacker',
			},
		},
	},
};
