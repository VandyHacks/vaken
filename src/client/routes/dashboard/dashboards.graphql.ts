import gql from 'graphql-tag';

export default gql`
	query myStatus {
		me {
			id
			... on Hacker {
				status
			}
		}
	}

	mutation confirmMySpot {
		confirmMySpot {
			id
			... on Hacker {
				status
			}
		}
	}

	mutation declineMySpot {
		declineMySpot {
			id
			... on Hacker {
				status
			}
		}
	}
`;
