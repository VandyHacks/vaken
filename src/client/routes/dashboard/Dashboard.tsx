import React, { FunctionComponent } from 'react';
import { Switch, Route } from 'react-router-dom';
import { HackerDash } from './HackerDash';
import { OrganizerDash } from './OrganizerDash';

export const Dashboard: FunctionComponent = (): JSX.Element => {
	return (
		<>
			<Switch>
				<Route path="/dashboard/organizer" component={OrganizerDash} />
				<Route path="/dashboard" component={HackerDash} />
			</Switch>
		</>
	);
};

export default Dashboard;
