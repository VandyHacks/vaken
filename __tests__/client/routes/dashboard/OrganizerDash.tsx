import React from 'react';
import renderer from 'react-test-renderer';
// import { ApolloProvider } from 'react-apollo';
// import { MockedProvider } from 'react-apollo/test-utils';
// import ApolloClient from 'apollo-boost';
import { SchoolTable } from '../../../../src/client/routes/dashboard/OrganizerDash';

// mock fetch
/*
global.fetch = jest.fn().mockImplementation(() =>
	Promise.resolve({
		ok: true,
		json: () => {
			message: 'hello world';
		},
	})
); */

it('Test SchoolTable renders properly', async () => {
	// see https://www.apollographql.com/docs/react/recipes/testing

	/*
	const mocks = [
		{
			request: {
				query: GET_STATISTICS,
				variables: { number: 5.0 },
			},
			result: {
				data: {
					getAllHackerStatuses: { id: '1' },
					getTopHackerSchools: { id: '1' },
					getAllHackerSizes: { id: '1' },
					getAllHackerGenders: { id: '1' },
				},
			},
		},
	];
    const client = new ApolloClient();
    */

	const component = renderer
		.create(<SchoolTable data={[{ counts: 1, school: 'mockschool' }]} />)
		.toJSON();
	expect(component).toMatchSnapshot();
});
