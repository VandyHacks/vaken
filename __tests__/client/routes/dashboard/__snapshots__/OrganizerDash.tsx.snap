// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizerDash 1`] = `
<ul
  className="sc-kpOJdX ibgOHi"
>
  <li
    className="sc-dxgOiQ cbKfUb"
  >
    <span
      className="sc-kgoBCf eCiLFY"
    >
      mockschool: 
    </span>
    <span
      className="sc-kGXeez lbwhzW"
    >
      1
    </span>
  </li>
</ul>
`;
